driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -665,7 +754,7 @@ public void shutdown() {
     }
 
     private static final class Flusher implements Runnable {
-        final EventLoop eventLoop;
+        final WeakReference<EventLoop> eventLoopRef;
```

Comment:
Another alternative to this would be to make `flusherLookup` also use `weakValues`.  I tested with this and it appeared to work exactly the same, but I think I like using `WeakReference` directly instead.  Any opinion or ideas to handle this better?